### DISCUSSION TITLE
Load in query data from a not related content

### DISCUSSION THREAD

**Discussion by StErMi** [prisma/prisma#15407] [Q&A] [answered]
At the moment I have this model (it's a partial view of the whole model)

```prisma
model ModelA {
  id           Int      @id @default(autoincrement())
  refNum       Int?
  refDate      String
  apiTimestamp DateTime
  value        Float
}

model ModelB {
  id       Int     @id @default(autoincrement())
  modelC   ModelC? @relation(fields: [modelCId], references: [id])
  modelCId Int?
  value    String
}

model ModelC {
  id          Int      @id @default(autoincrement())
  modelBArray ModelB[]
  timestamp   DateTime @default(now())
}
```

Now what I would like to achieve is this

```typescript
const modelCList = await prisma.modelC.findMany({
  include: {
      modelBs: true,
      modelA: {
          where: {
              refDate: 'internalValueOfModelC.timestamp formatted as as MM/dd/yyyy',
          },
      },
  },
});
```

I have two problem shere:

- in the `include` part I would like to subload a model that as you can see has no direct relationship. `ModelC` and `ModelA` have no relation
- in the `where` clause reference the "internal" value of `modelC.timestamp`
- the referenced value of `modelC.timestamp` is a `Datetime` but must be formatted as `MM/dd/yyyy`

Is it possible to do that in Prisma?

**Comment by nikolasburk** [comment, marked as answer]
> in the `include` part I would like to subload a model that as you can see has no direct relationship. `ModelC` and `ModelA` have no relation

If none of the models have a relation to `ModelA`, how do you expect Prisma to behave in loading the relation via `include`? 🤔 

If you want to load all `ModelA` records based on a filter, this sounds like you should be using a plain `modelA.findMany` query with a `where` filter since there's no aspect of a relation that would justify `include` (at least as far as I can tell from your schema): 

```ts
const modelCList = await prisma.modelC.findMany({
  include: {
    modelBArray: true,
  },
});
const modelAList = await prisma.modelA.findMany({
  where: {
    refDate: {
      equals: ...
    }
  }
})
```

If you're concerned about two separate queries and that's the reason why you want to "subload" the `ModelA` records, you can batch the two queries in a transactions:

```ts
const modelCListQuery = prisma.modelC.findMany({
  include: {
    modelBArray: true,
  },
});
const modelAListQuery = prisma.modelA.findMany()
const [modelCList, modelAList] = await prisma.$transaction([modelCListQuery, modelAListQuery])
```

> in the `where` clause reference the "internal" value of `modelC.timestamp`
> the referenced value of `modelC.timestamp` is a `Datetime` but must be formatted as MM/dd/yyyy

Hmm, that's a good question! I don't think that's possible out of the box tbh but can't tell for sure... Prisma Client returns `DateTime` as native `Date` objects and follow the ISO 8601 standard for formatting.
